android/jni/com/mapswithme/maps/Framework.cpp

```diff
@@ -891,6 +892,51 @@ Java_com_mapswithme_maps_Framework_nativeGetRouteFollowingInfo(JNIEnv * env, jcl
   return result;
 }
 
+JNIEXPORT jintArray JNICALL
+Java_com_mapswithme_maps_Framework_nativeGenerateRouteAltitudeChartBits(JNIEnv * env, jclass, jint width, jint height)
+{
+  ::Framework * fr = frm();
+  ASSERT(fr, ());
+  vector<uint8_t> imageRGBAData;
+
+  if (!fr->GenerateRouteAltitudeChart(width, height, imageRGBAData))
+  {
+    LOG(LWARNING, ("Cann't generate route altitude image."));
+    return nullptr;
+  }
+
+  size_t const imageRGBADataSize = imageRGBAData.size();
+  if (imageRGBADataSize == 0)
+  {
+    LOG(LWARNING, ("GenerateRouteAltitudeChart returns true but the vector with altitude image bits is empty."));
+    return nullptr;
+  }
+
+  size_t const pxlCount = width * height;
+  if (maps::kAlitudeChartBPP * pxlCount != imageRGBAData.size())
+  {
+    LOG(LWARNING, ("Wrong size of vector with altitude image bits. Expected size:", pxlCount, ". Real size:", imageRGBAData.size()));
+    return nullptr;
+  }
+
+  jintArray imageRGBADataArray = env->NewIntArray(imageRGBADataSize);
+  ASSERT(imageRGBADataArray, ());
+  jint * arrayElements = env->GetIntArrayElements(imageRGBADataArray, 0);
+  ASSERT(arrayElements, ());
+
+  for (size_t i = 0; i < imageRGBADataSize; ++i)
+  {
+    size_t const shiftInBytes = i * maps::kAlitudeChartBPP;
+    arrayElements[i] = (imageRGBAData[shiftInBytes + 3] << 24) /* alpha */
+        | (imageRGBAData[shiftInBytes] << 16) /* red */
```

Comment:
As far as I'm aware uint8_t should be promoted up to int before shifting. (Is not it so?)
arrayElements is an array of jint (typedef on int). 

So if the size of int is more or equal 4 bytes everything is ok. I will:
1. Add static_assert(sizeof(jint) >= 4, "Size of jint in less than 4 bytes.");
2. add static_cast
```
    arrayElements[i] = (static_cast<jint>(imageRGBAData[shiftInBytes + 3]) << 24) /* alpha */
        | (static_cast<jint>(imageRGBAData[shiftInBytes]) << 16) /* red */
        | (static_cast<jint>(imageRGBAData[shiftInBytes + 1]) << 8) /* green */
        | (static_cast<jint>(imageRGBAData[shiftInBytes + 2])); /* blue */
```